lms/services/group_info.py

```diff
@@ -1,66 +1,48 @@
-"""A service that upserts :class:`lms.models.GroupInfoService` records."""
+"""A service for managing `GroupInfo` records."""
 
-from lms.models import GroupInfo
-
-__all__ = ["GroupInfoService"]
+from lms.models import GroupInfo, Grouping
 
 
 class GroupInfoService:
-    """
-    A service that upserts :class:`~lms.models.GroupInfo` records.
-
-    Usage::
+    """A service for managing `GroupInfo` records."""
 
-        group_info = request.find_service(name="group_info")
-        group_info.upsert(h_group, application_instance, request.params)
-    """
+    def __init__(self, _context, request):
+        self._db = request.db
+        self._lti_user = request.lti_user
 
-    GROUPING_TYPES = {
+    _GROUPING_TYPES = {
         "course": "course_group",
         "canvas_section": "section_group",
         "canvas_group": "canvas_group_group",
         "blackboard_group": "blackboard_group_group",
     }
 
-    def __init__(self, _context, request):
-        self._db = request.db
-        self._lti_user = request.lti_user
-
-    def upsert(self, h_group, application_instance, params: dict):
+    def upsert_group_info(self, grouping: Grouping, params: dict):
         """
         Upsert a row into the `group_info` DB table.
 
-        Find the models.GroupInfo matching the given h_group or create it if
-        none exists. Then update the GroupInfo's application_instance to the given
-        application_instance, and update its other columns from the items in `params`.
-
-        params["id"], params["authority_provided_id"], and params["info"] will
-        be ignored if present--these columns can't be updated.
-
-        Any keys in `params` that don't correspond to a GroupInfo column name
-        will be ignored.
-
-        :param h_group: the group to upsert
-        :type h_group: models.HGroup
-
-        :param application_instance: the ApplicationInstance this group belongs to
-
-        :param params: the other GroupInfo columns to set
```

Comment:
This feels like it's just giving a blow by blow of the code rather than an overview of things you need to know.